src/sentry/api/event_search.py

```diff
@@ -194,10 +204,16 @@ def visit_raw_search(self, node, children):
 
     def visit_numeric_filter(self, node, children):
         search_key, _, operator, search_value = children
-        try:
-            search_value = int(search_value.text)
-        except ValueError:
-            raise InvalidSearchQuery('Invalid numeric query: %s' % (search_key,))
+        # Tags are always strings
```

Comment:
I'm pretty sure this is true?

Reply:
Yes.

Reply:
I think we probably need to add in a better layer that says which fields can be which types. At the moment we'll get errors if we use a date format on a field that doesn't support it (I think). This will help a little, but I think we can probably be more declarative and just whitelist valid fields.